File: src/main/java/com/linebot/model/qiita/Item.java
package com.linebot.model.qiita;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Item {
    // https://qiita.com/api/v2/docs#get-apiv2usersuser_iditems
    // いろんなデータがあるが、いったん欲しい情報だけ抽出する
    // 記載順序も変えている

    private String id;
    private String title;
    private String url;

    private boolean coediting;
    @JsonProperty("comments_count")
    private int commentsCount;
    @JsonProperty("likes_count")
    private int likesCount;
    @JsonProperty("page_views_count")
    private int pageViewsCount;
    @JsonProperty("reactions_count")
    private int reactionsCount;

    @JsonProperty("created_at")
    private Date createdAt;
    @JsonProperty("updated_at")
    private Date updatedAt;
}

File: src/main/kotlin/com/linebot/model/qiita/Item.kt
package com.linebot.model.qiita

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.Date

data class Item(
    // https://qiita.com/api/v2/docs#get-apiv2usersuser_iditems
    // いろんなデータがあるが、いったん欲しい情報だけ抽出する
    // 記載順序も変えている
    var id: String? = null,
    var title: String? = null,
    var url: String? = null,
    var coediting: Boolean = false,
    @JsonProperty("comments_count")
    var commentsCount: Int = 0,
    @JsonProperty("likes_count")
    var likesCount: Int = 0,
    @JsonProperty("page_views_count")
    var pageViewsCount: Int = 0,
    @JsonProperty("reactions_count")
    var reactionsCount: Int = 0,
    @JsonProperty("created_at")
    var createdAt: Date? = null,
    @JsonProperty("updated_at")
    var updatedAt: Date? = null
)
